### PR TITLE
task form body

### PR DESCRIPTION
**Pull request by Catalin12**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+import { DialogModule } from "primeng/dialog";
+import { ReactiveFormsModule } from "@angular/forms";
+
+import { ButtonModule } from "primeng/button";
+import { BrowserAnimationsModule } from "@angular/platform-browser/animations";
+
 import { HomePageComponent } from "./home-page/home-page.component";
+import { TaskFormComponent } from "./task-form/task-form.component";
+
+
```
Comment (carmina-c) [inline src/app/app.module.ts]: i think lines 8, 14, 15 should be deleted

File: src/app/task-form/task-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+})
+export class TaskFormComponent {
+
+	public taskForm = this.formBuilder.group({
+		title: ["", Validators.required],
+		description: [""],
+		isCompleted: false
+	});
+	public name = new FormControl("");
+	public constructor(
```
Comment (carmina-c) [inline src/app/task-form/task-form.component.ts:17]: maybe a new line before constructor for readability?

File: src/app/home-page/home-page.component.ts [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
 
 @Component({
 	selector: "home-page",
 	templateUrl: "./home-page.component.html",
 	styleUrls: ["./home-page.component.css"]
 })
-export class HomePageComponent implements OnInit {
+export class HomePageComponent {
 
+	// eslint-disable-next-line @typescript-eslint/no-empty-function
```
Comment (danielursu) [inline src/app/home-page/home-page.component.ts]: check eslint constructor exception

File: src/app/task-form/task-form.component.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<form [formGroup]="taskForm" (ngSubmit)="onSubmit()">
+
+	<span class="p-float-label" style="margin-bottom:15px">
+		<input type="text" id="title" formControlName="title" placeholder="Task Titl" pInputText>
+		<label *ngIf="!taskForm.value.title" for="inputtext">Task Title</label>
```
Comment (danielursu) [inline src/app/task-form/task-form.component.html]:
```suggestion
		<input type="text" id="title" formControlName="title" pInputText>
		<label for="title">Task Title</label>
```
if you have floating label, you do not need placeholder anymore

File: src/app/task-form/task-form.component.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+	</span>
+
+	<span class="p-float-label" style="margin-bottom:15px">
+		<input type="text" id="description" formControlName="description" pInputText>
+		<label *ngIf="!taskForm.value.description" for="inputtext">Task Description</label>
```
Comment (danielursu) [inline src/app/task-form/task-form.component.html]: same

File: src/app/task-form/task-form.component.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+	</span>
+	<button pButton type="submit" class="p-button-success" style="margin:5px"
+		[disabled]="!taskForm.value.title">Submit</button>
```
Comment (danielursu) [inline src/app/task-form/task-form.component.html]:
```suggestion
		[disabled]="!taskForm.valid">Submit</button>
```
check for validation of the entire form

File: src/app/task-form/task-form.component.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+	styleUrls: ["./task-form.component.css"]
+})
+export class TaskFormComponent {
+
+	public taskForm = this.formBuilder.group({
+		title: ["", Validators.required],
+		description: [""],
+		isCompleted: false
+	});
+	public name = new FormControl("");
```
Comment (danielursu) [inline src/app/task-form/task-form.component.ts]:
```suggestion
	public name = new FormControl("");
```
do you need this?

**Review by danielursu — CHANGES_REQUESTED** [review]

File: src/app/app.module.ts (hunk 1 above)
Comment (Catalin12) [reply src/app/app.module.ts]: better now, ty

File: src/app/task-form/task-form.component.html (hunk 4 above)
Comment (Catalin12) [reply src/app/task-form/task-form.component.html]: Before
image


After
image

I think that ngIf should be left there

File: src/app/task-form/task-form.component.html (hunk 6 above)
Comment (Catalin12) [reply src/app/task-form/task-form.component.html]: I remember that only the title is required the description is not.

File: src/app/task-form/task-form.component.ts (hunk 7 above)
Comment (Catalin12) [reply src/app/task-form/task-form.component.ts]: sorry, it remained from when I was doing research

File: src/app/task-form/task-form.component.html (hunk 4 above)
Comment (danielursu) [reply src/app/task-form/task-form.component.html]: without ngIf if you choose floating label approach
check the docs https://www.primefaces.org/primeng/inputtext

File: src/app/task-form/task-form.component.html (hunk 6 above)
Comment (danielursu) [reply src/app/task-form/task-form.component.html]: what if you add another required form control?
better approach is to verify entire form

File: src/app/task-form/task-form.component.html (hunk 4 above)
Comment (danielursu) [reply src/app/task-form/task-form.component.html]: see also the source tab from the link

File: src/app/task-form/task-form.component.html (hunk 6 above)
Comment (Catalin12) [reply src/app/task-form/task-form.component.html]: i got it

**Review by danielursu — APPROVED** [review]

**Pull request merged**